collectors/cveorg/collectors.py

```diff
@@ -0,0 +1,328 @@
+        """
+        Collect vulnerability data from the cvelistV5 repository and store them in OSIDB.
+        """
+        if not self.snippet_creation_enabled:
+            msg = "Snippet creation is disabled. The CVEorg collector is not running."
+            logger.error(msg)
+            return msg
+
+        # Set osidb.acl to be able to CRUD database properly and essentially bypass ACLs as
+        # Celery workers should be able to read/write any information in order to fulfill their jobs
+        set_user_acls(settings.ALL_GROUPS)
+
+        logger.info("Starting CVEorg data collection.")
+        new_snippets = []
+        new_flaws = []
+
+        self.clone_repo()
+        self.update_repo()
+        changes, period_end = self.get_repo_changes()
+
+        file_paths = set(
+            f"{self.REPO_PATH}/{f}"
+            for f in changes.split("\n")
+            if re.search(self.CVE_PATH, f)
+        )
+        for file_path in file_paths:
+            with open(file_path) as file:
+                file_content = json.load(file)
+                if file_content["cveMetadata"]["state"] == "PUBLISHED":
+                    try:
+                        content = self.extract_content(file_content)
+                    except Exception as exc:
+                        msg = f"Failed to parse data from the {file_path} file: {exc}"
+                        logger.error(msg)
+                        continue
+
+                    try:
+                        with transaction.atomic():
+                            new_snippet, new_flaw = self.save_snippet_and_flaw(content)
+                            if new_snippet:
+                                new_snippets.append(content["cve_id"])
+                            if new_flaw:
+                                new_flaws.append(content["cve_id"])
+                    except Exception as exc:
+                        message = f"Failed to save snippet and flaw for {content['cve_id']}. Error: {exc}."
+                        logger.error(message)
+                        raise CVEorgCollectorException(message) from exc
+
+        updated_until = period_end
+        self.store(complete=True, updated_until_dt=updated_until)
+        msg = (
+            f"{self.name} is updated until {updated_until}."
+            f"Created snippets: {', '.join(new_snippets) if new_snippets else 'none'}. "
+            f"Created CVEs from snippets: {', '.join(new_flaws) if new_flaws else 'none'}."
+        )
+        logger.info("CVEorg sync was successful.")
+        return msg
+
+    def save_snippet_and_flaw(self, content: dict) -> tuple[bool, bool]:
+        """
+        Create and save snippet with flaw from normalized `content` if they do not already exist.
+        The creation is restricted by the published date and allowed keywords.
+        """
+        snippet_created = False
+        flaw_created = False
+
+        if self.snippet_creation_start_date and content["unembargo_dt"]:
```

Comment:
`unembargo_dt` (i.e. published date) can be missing. In these cases, a flaw is created to not miss anything important.